.github/workflows/ci-test-ginkgo.yml

```
@@ -48,14 +48,21 @@ jobs:
       - name: Install an annotation controller
         run: ./.github/workflows/install-annotation-controller.sh
 
-      - name: Run kubectl proxy
-        run: kubectl proxy &
+      - name: Generate KubeArmor artifacts
+        run: |
+          GITHUB_SHA=$GITHUB_SHA ./KubeArmor/build/build_kubearmor.sh
+          if [ ${{ matrix.runtime }} != "docker" ]; then
+            # Import Images to k3s Registry
+            docker save kubearmor/kubearmor-init:latest | sudo k3s ctr images import -
+            docker save kubearmor/kubearmor:latest | sudo k3s ctr images import -
+          fi
 
-      - name: Run KubeArmor in background
+      - name: Run KubeArmor
         run: |
-          make -C KubeArmor/ run &
-          bash -c 'for((i=0;i<300;i++)); do cnt=$(ps -e | grep kubearmor | wc -l); [[ $cnt -gt 0 ]] && echo "Kubearmor is running" && break; sleep 1; done'
-        timeout-minutes: 5
+          kubectl apply -f deployments/CRD
+          kubectl apply -f ./KubeArmor/build/kubearmor-test-${{ matrix.runtime }}.yaml
```

@daemon1024 we might have inconsistency in these runtime specific manifests.

---

They seem to be the same.

---

When RUNTIME is containerd, the scripts will install k3s with the default embedded containerd the sock file for which is at `/run/k3s/containerd/containerd.sock`. However the kubearmor-test-containerd manifest mounts the default containerd sock file.
```suggestion
          if [ ${{ matrix.runtime }} == "containerd" ]; then
            kubectl apply -f ./KubeArmor/build/kubearmor-test-k3s.yaml
          else
            kubectl apply -f ./KubeArmor/build/kubearmor-test-${{ matrix.runtime }}.yaml
          fi
```